protocol/contracts/beanstalk/sun/SeasonFacet/Weather.sol

```diff
@@ -193,29 +283,18 @@ contract Weather is Sun {
      * and become Harvestable.
      * For more information On Oversaturation see {Weather.handleRain}.
      */
-    function sop() private {
-        // calculate the beans from a sop.
-        // sop beans uses the min of the current and instantaneous reserves of the sop well,
-        // rather than the twaReserves in order to get bean back to peg.
-        address sopWell = s.sopWell;
-        (uint256 newBeans, IERC20 sopToken) = calculateSop(sopWell);
-        if (newBeans == 0) return;
-
-        uint256 sopBeans = uint256(newBeans);
-        uint256 newHarvestable;
-
-        // Pay off remaining Pods if any exist.
-        if (s.f.harvestable < s.r.pods) {
-            newHarvestable = s.r.pods - s.f.harvestable;
-            s.f.harvestable = s.f.harvestable.add(newHarvestable);
-            C.bean().mint(address(this), newHarvestable.add(sopBeans));
-        } else {
-            C.bean().mint(address(this), sopBeans);
-        }
+    function sop(WellDeltaB memory wellDeltaB) private {
+        if (wellDeltaB.reductionAmount == 0) return;
+        IERC20 sopToken = LibWell.getNonBeanTokenFromWell(wellDeltaB.well);
+
+        uint256 sopBeans = wellDeltaB.reductionAmount;
+
+        // code reviewer note: pre-calc total amount of beans to mint and mint them all at once?
+        C.bean().mint(address(this), sopBeans);
```

Comment:
pre-calculating would most likely be more gas efficent, but this is cleaner. Defer to you on implmentation (I'd probably just go with this).

Reply:
I like current way, will leave as-is